cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/resource/AbstractOperations.java

```diff
@@ -185,38 +186,49 @@ public T get(String namespace, String name) {
-                        } else {
-                            future.fail("Not ready yet");
+        long deadline = System.currentTimeMillis() + timeoutMs;
+
+        Handler<Long> handler = new Handler<Long>() {
+            @Override
+            public void handle(Long timerId) {
+
+                vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                    future -> {
+                        try {
+                            if (isReady(namespace, name))   {
+                                future.complete();
+                            } else {
+                                if (log.isTraceEnabled()) {
+                                    log.trace("{} {} in namespace {} is not ready", resourceKind, name, namespace);
+                                }
+                                future.fail("Not ready yet");
+                            }
+                        } catch (Exception e) {
+                            log.warn("Caught exception while waiting for {} {} in namespace {} to get ready", resourceKind, name, namespace, e);
+                            future.fail(e);
                         }
-                    } catch (Exception e) {
-                        log.warn("Caught exception while waiting for {} {} in namespace {} to get ready", resourceKind, name, namespace, e);
-                        future.fail(e);
-                    }
-                },
-                false,
-                res -> {
-                    if (res.succeeded())    {
-                        vertx.cancelTimer(timerId);
-                        log.info("{} {} in namespace {} is ready", resourceKind, name, namespace);
-                        fut.complete();
-                    } else {
-                        if (System.currentTimeMillis() - startTime > timeoutMs)   {
-                            vertx.cancelTimer(timerId);
-                            log.error("Exceeded timeoutMs of {} ms while waiting for {} {} in namespace {} to be ready", timeoutMs, resourceKind, name, namespace);
-                            fut.fail(new TimeoutException());
+                    },
+                    false,
+                    res -> {
+                        if (res.succeeded()) {
+                            log.info("{} {} in namespace {} is ready", resourceKind, name, namespace);
+                            fut.complete();
+                        } else {
+                            long timeLeft = deadline - System.currentTimeMillis();
+                            if (timeLeft <= 0) {
+                                log.error("Exceeded timeoutMs of {} ms while waiting for {} {} in namespace {} to be ready", timeoutMs, resourceKind, name, namespace);
+                                fut.fail(new TimeoutException());
+                            }
+                            // Schedule ourselves to run again
+                            vertx.setTimer(Math.min(pollIntervalMs, timeLeft), this);
```

Comment:
This will set a new timer even when the time timeLeft is gone, or? Should't this be in the `else` section for the `if (timeLeft <= 0)`?

Reply:
You're right.